### PR TITLE
v20: Add transaction-related types and messages

### PR DESCRIPTION
**Pull request by romainletendart**
Basic transactions as described in OCPP2.0 Section E require certain messages and types to be available hence this pull request.

### CODE REVIEW

File: ocpp_codec/v20/messages.py [hunk 1]
```diff
@@ -117,6 +117,15 @@ class Response:
         setVariableResult: typing.List[types.SetVariableResultType]
 
 
+class StatusNotification(Action):
+    @dataclass
+    class Request:
```
Comment (rsebille) [inline ocpp_codec/v20/messages.py:122]: Why only add `Request`?
If we add a new message we should add the _Request_ and the _Response_, both will be needed at one point.

File: ocpp_codec/validators.py [hunk 2]
```diff
@@ -63,6 +63,14 @@ def noop(value: typing.Any) -> typing.Any:
     return value
 
 
+def min_length(length: int, value: typing.Any) -> typing.Any:
+    """Validates that an input doesn't go below a given length."""
+    actual_length = len(value)
+    if actual_length < length:
+        raise errors.PropertyConstraintViolationError("Input is too short", min_length=length)
```
Comment (rsebille) [inline ocpp_codec/validators.py]: Maybe add the input and/or `actual_length` to the log so we know exactly why that went wrong?

File: ocpp_codec/v20/messages.py (hunk 1 above)
Comment (romainletendart) [reply ocpp_codec/v20/messages.py:122]: `Action` already defines a `Response` that `StatusNotification` inherits and the specification reads `No fields are defined.` for `StatusNotificationResponse`.

File: ocpp_codec/validators.py (hunk 2 above)
Comment (romainletendart) [reply ocpp_codec/validators.py]: It seems required list fields must have at least one element to be considered defined.
As such, we do not need this validator anymore so I just removed it.
I took into account your comment and applied it to `max_length` validator which has the same issue.

File: ocpp_codec/v20/messages.py (hunk 1 above)
Comment (romainletendart) [reply ocpp_codec/v20/messages.py:122]: For various reasons (logging, comparison), I ended up overriding `Response` with an empty method just to have it defined.

Comment (lburg) [reply ocpp_codec/v20/messages.py:122]: Yes this is the right way to go with the current implementation. The `Action.Response` is to be taken as some sort of example, we should rework this a bit.

File: ocpp_codec/v20/types.py [hunk 3]
```diff
@@ -97,6 +104,42 @@ class IdTokenEnum(utils.AutoNameEnum):
...
+class MeasurandEnum(utils.AutoNameEnum):
+    CurrentExport = 'Current.Export'
+    CurrentImport = 'Current.Import'
+    CurrentOffered = 'Current.Offered'
+    EnergyActiveExportRegister = 'Energy.Active.Export.Register'
+    EnergyActiveImportRegister = 'Energy.Active.Import.Register'
+    EnergyReactiveExportRegister = 'Energy.Reactive.Export.Register'
+    EnergyReactiveImportRegister = 'Energy.Reactive.Import.Register'
+    EnergyActiveExportInterval = 'Energy.Active.Export.Register'
+    EnergyActiveImportInterval = 'Energy.Active.Import.Register'
```
Comment (lburg) [inline ocpp_codec/v20/types.py]:
```suggestion
    EnergyActiveExportInterval = 'Energy.Active.Export.Interval'
    EnergyActiveImportInterval = 'Energy.Active.Import.Interval'
```

File: ocpp_codec/validators.py [hunk 4]
```diff
@@ -67,7 +67,9 @@ def max_length(length: int, value: typing.Any) -> typing.Any:
     """Validates that an input doesn't exceed a given length."""
     actual_length = len(value)
     if actual_length > length:
-        raise errors.PropertyConstraintViolationError("Input is too long", max_length=length)
+        raise errors.PropertyConstraintViolationError(
```
Comment (lburg) [inline ocpp_codec/validators.py:70]: Good idea, but this should be in its own commit

**Review by lburg — COMMENTED** [review]
A few comments but nothing big, thanks you for contributing :)

File: ocpp_codec/validators.py (hunk 4 above)
Comment (romainletendart) [reply ocpp_codec/validators.py:70]: Done

**Review by lburg — APPROVED** [review]
LGTM :+1:
